### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by nleek0**
Name: Nathan Leeks

Student Number: 47440599

Description: The purpose of the model was to upscale images of brain mri with minimum loss of detail. The dataset chosen was the ADNI dataset, and the model chosen was the Efficient Sub-pixel CNN.

The model was able produce reasonably clear images with minimum loss of detail.

Project structure:
modules.py - This file contained the model used to train the data set.

dataset.py - This file contains all the preprocessing needed for the training.

train.py - This file contains all the code needed to train the model and saves a saved state of the model after it is trained.

predict.py - This file loads the saved state of the model and then displays example outputs of the model.

utils.py - This file contains all the constants that are used in the other files. 

README.md - This file contains the documentation of the training, results, and the model.

### CODE REVIEW

**Comment by uqmarlonbran** [comment]
**This is an initial inspection - no action is required yet**

**Difficulty:** Normal

**General Comments:** Overall a good attempt. Use of a validation set would be beneficial. Also a network architecture visual that accurately represents the coded network. Image quality seems in-line with other attempts.

**[Recognition Problem]**
"modules.py":

- Re-implementation of the Keras example using PyTorch. All looks ok.

"dataset.py":

- Nice use of ImageLoader to deal with NC and AD. All seems ok.

"train.py":

- Doesn't use a validation set.
- Training loop looks ok, correct tracking of training loss.
- Simply stores the final trained network at the end.

"predict.py":

- Correctly loads the trained weights.
- Provides some visualisations of the output images.
- No metrics on the test set.

"README.MD":

- Graphic of model says Tanh, though their code uses relu activation functions.
- Brief description of network architecture and approach.
- Claims to use validation set, however the provided training figure and training script do not create a validation set.
- Training looks ok, input/output images look ok.
- Dependencies provided.

**[Commit Log]**
Commit log seems ok.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no validation set -1
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
